### PR TITLE
Gutenberg: add a section for development

### LINKED ISSUE #25971: Add a Gutenberg section, with it's own separate js and css bundles
Add a /gutenberg section, with it's own separate js and css bundles. This should be hidden behind it's own feature flag.

### PR DESCRIPTION
**Pull request by gwwar**
This closes #25971, and adds a feature flag and basic routes: `/gutenberg` and `/gutenberg/post` and `/gutenberg/page` to allow us to work on adding Gutenberg into Calypso without a long lived feature branch. Eventually when Gutenberg will take over as the primary editor, it will use `/post` and `/page`, but that is a ways off.

@jsnajdr has made an excellent start in https://github.com/Automattic/wp-calypso/pull/25590 and @Automattic/lannister will be building on top of that work, and pulling in smaller changes to master, in the next few weeks.

Care was taken to add a new css and js bundle so we don't bloat other sections during development.

Folks may verify that no js bundles were affected here:
http://iscalypsofastyet.com/branch?branch=add/gutenberg-section

#### With the gutenberg feature flag enabled:
- Start calypso in a development environment with `npm start`
- Navigate to /gutenberg
- You are redirected to /gutenberg/post and prompted to select a site
- Site Selection works and we see a hello world page:
screen shot 2018-07-11 at 6 12 07 pm

#### With the gutenberg feature flag disabled
- Stop the Calypso instance and run with the `gutenberg` feature flag disabled `DISABLE_FEATURES=gutenberg npm start`
-  Navigate to /gutenberg
- You are redirected to /post
(Note that new routes need to be whitelisted for WordPress.com anyway)

#### Verify
- Verify that the gutenberg styles are in their own css file.
- Verify that we do not bloat other js sections
- No Calypso Regressions

### Next Steps After this PR lands
- Some tasks I see after this is figuring out a good working space for the editor. For example, can we avoid rendering the masterbar, and adding additional css resets?
- In parallel, we can experiment with pulling in needed Gutenberg files without copying them physically. For example, either by helping publish additional packages, or using github dep links.

### CODE REVIEW

File: client/gutenberg-editor/index.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+export default function() {
+	if ( config.isEnabled( 'gutenberg' ) ) {
+		page( '/gutenberg', () => page.redirect( '/gutenberg/post' ) );
+		page( '/gutenberg/post', siteSelection, sites, makeLayout, clientRender );
+		page( '/gutenberg/post/:site?/:post?', siteSelection, post, makeLayout, clientRender );
+		page( '/gutenberg/page', siteSelection, sites, makeLayout, clientRender );
+		page( '/gutenberg/page/:site?/:post?', siteSelection, post, makeLayout, clientRender );
+	} else {
+		page( '/gutenberg', () => page.redirect( '/post' ) );
+		page( '/gutenberg/*', () => page.redirect( '/post' ) );
```
Comment (jsnajdr) [inline client/gutenberg-editor/index.js]: Nit: redirect can be written as `page( '/gutenberg', '/post' )`.

File: client/gutenberg-editor/main.jsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+ */
+import React, { Component } from 'react';
+
+class GutenbergEditor extends Component {
+	render() {
+		return <div className="gutenberg-editor">Hello, Gutenberg!</div>;
+	}
+}
+
+export default GutenbergEditor;
```
Comment (dmsnell) [inline client/gutenberg-editor/main.jsx]: directory structure seems reasonable. the other idea is to bundle all Gutenberg-related content inside `client/gutenberg` making this `client/gutenberg/editor` and others `client/gutenberg/extensions` or something like that.

File: client/gutenberg-editor/index.js (hunk 1 above)
Comment (gwwar) [reply client/gutenberg-editor/index.js]: Good catch, that was a copy pasta on my part. Updated in 76d3e27

**Comment by gwwar** [comment]
> making this client/gutenberg/editor and others client/gutenberg/extensions or something like that.

@dmsnell I'll give it a shot and see if I run into trouble.

File: client/gutenberg-editor/main.jsx (hunk 2 above)
Comment (gwwar) [reply client/gutenberg-editor/main.jsx]: Thanks updated in 45b203b!

**Comment by gwwar** [comment]
Thanks for the reviews @jsnajdr @dmsnell !

**Pull request merged**